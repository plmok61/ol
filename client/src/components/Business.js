import React, { Component } from 'react'
import axios from 'axios'

export default class Business extends Component {
  constructor (props) {
    super(props)
    this.state = {
      business: false
    }
  }

  componentDidMount () {
    axios.get(`http://ec2-54-84-251-148.compute-1.amazonaws.com/businesses/${this.props.params.id}`)
      .then(res => {
        this.setState({
          business: res.data
        })
        console.log(res.data)
      })
  }

  render () {
    const { name, address, address2, city, country, phone, state, website, zip } = this.state.business
    
    if (this.state.business) {
      return (
        <div>
          {name}
        </div>
      )
    } else {
      return <div>loading</div>
    }
  }
}