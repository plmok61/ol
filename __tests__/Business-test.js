import React from 'react'
import Business from '../client/src/components/Business'
import renderer from 'react-test-renderer'

test('Business renders', () => {
  const component = renderer.create(
    <Business params={{id:1}}/>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})