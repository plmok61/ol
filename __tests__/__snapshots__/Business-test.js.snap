exports[`test Business renders 1`] = `
<div>
  loading
</div>
`;

exports[`test test 2 1`] = `
<div>
  loading
</div>
`;
